### PR TITLE
WIP: Changing/Omitting tests to decrease size of downloaded datasets

### PR DESCRIPTION
**Pull request by pritamd47**
Currently,
```[Julia]
test ArchGDAL
````
downloads data-sets amounting to roughly 200MB, which inadvertently adds to the time required to test the package. 

As discussed in issue #104, there are some simple fixes to this problem, which have been implemented in these commits. 
- Removed `Aster.img`, and replaced with a cropped version of the same `Cropped_aster.img`
- Rewrote some tests (made minor changes to values such as count and sum of pixels, verifying with rasterio and QGIS), and commented out some which require a combination of data-sets including the original `Aster.img` and some other feature data-sets. This has decreased the code-coverage and hence some alternate tests have to be written in place of those commented out. 

As suggested by @evetion [here](https://github.com/yeesian/ArchGDAL.jl/issues/104#issuecomment-578207816), I am opening this PR for review. 

This is my first PR to any open-source project, and I am sure there will be mistakes somewhere (I might have tinkered with the CI stuff, while not knowing how they work, and might've broken things), hence, please feel free to guide me in the right direction if you notice some mistakes, irregularities or deviations (from conventions) or anything for that matter – I would be really grateful!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,8 +7,11 @@
 docs/build/
 docs/site/
 
+# Commenting out these lines for testing and coverage
```
[the remark this answers is not in the document]
Comment (pritamd47) [reply .gitignore]: Removed.
 9940051

File: .gitignore [hunk 2]
```diff
@@ -7,8 +7,11 @@
...
 test/data/
 test/gdalworkshop/
 test/ospy/
 test/pyrasterio/
 test/spatialite/
+
+.vscode/
```
[the remark this answers is not in the document]
Comment (pritamd47) [reply .gitignore]: Fixed. Configured a global gitignore file as well. I didn't know this existed!
9940051

File: .codecov.yml [hunk 3]
```diff
@@ -1 +0,0 @@
-comment: false
```
[the remark this answers is not in the document]
Comment (pritamd47) [reply .codecov.yml]: Added again. It was indeed unintentional. This is my first time using Travis and codecov, and I couldn't exactly figure out how to get these working. I guess I have to read about them thoroughly first.
 e9385c9

File: Manifest.toml [hunk 4]
```diff
@@ -0,0 +1,101 @@
+# This file is machine-generated - editing it directly is not advised
```
[the remark this answers is not in the document]
Comment (pritamd47) [reply Manifest.toml]: Removed e9385c92fb0165e87fbae24e2affa7fdf28ff56b and added to gitignore 6a80a4a9aecf1d630d7ff4ec3344e3476c776436

File: README.md [hunk 5]
```diff
@@ -1,8 +1,8 @@
 # ArchGDAL
 
-[![Build Status](https://travis-ci.org/yeesian/ArchGDAL.jl.svg?branch=master)](https://travis-ci.org/yeesian/ArchGDAL.jl)
-[![Build status](https://ci.appveyor.com/api/projects/status/github/yeesian/ArchGDAL.jl?svg=true&branch=master)](https://ci.appveyor.com/project/NgYeeSian/archgdal-jl/branch/master)
-[![Coverage Status](https://coveralls.io/repos/github/yeesian/ArchGDAL.jl/badge.svg?branch=master)](https://coveralls.io/github/yeesian/ArchGDAL.jl?branch=master)
+[![Build Status](https://travis-ci.com/pritamd47/ArchGDAL.jl.svg?branch=master)](https://travis-ci.com/pritamd47/ArchGDAL.jl)
+[![Build status](https://ci.appveyor.com/api/projects/status/github/pritamd47/ArchGDAL.jl?svg=true&branch=master)](https://ci.appveyor.com/project/pritamd47/archgdal-jl/branch/master)
+[![Coverage Status](https://coveralls.io/repos/github/pritamd47/ArchGDAL.jl/badge.svg?branch=master)](https://coveralls.io/github/pritamd47/ArchGDAL.jl?branch=master)
```
[the remark this answers is not in the document]
Comment (pritamd47) [reply README.md]: Updated 994005162240bd2693a5096286fa3c191f2b541f

File: test/runtests.jl [hunk 6]
```diff
@@ -21,6 +21,8 @@ include("remotefiles.jl")
         include("test_dataset.jl")
         include("test_rasterband.jl")
         include("test_rasterio.jl")
+
+        # Uses largest raster
```
[the remark this answers is not in the document]
Comment (pritamd47) [reply test/runtests.jl]: Removed the unnecessary white spaces and comments. Kept some comments explaining how the values were calculated. f95ccfaec7a816376110c39311211be3284dbbd5

File: test/test_rasterio.jl [hunk 7]
```diff
@@ -2,135 +2,136 @@ using Test
...
-        for (cols,rows) in AG.windows(band)
-            AG.rasterio!(ds, buffer, Cint[1], rows .- 1, cols .- 1)
-            data = buffer[1:length(cols),1:length(rows)]
-            count += sum(data .> 0)
-            total += sum(data)
-        end
-        @test total / count ≈ 76.33891347095299
-        @test total / (AG.height(ds) * AG.width(ds)) ≈ 47.55674749653172
-    end
+# AG.read("ospy/data4/cropped_aster.img") do ds
```
[the remark this answers is not in the document]
Comment (pritamd47) [reply test/test_rasterio.jl]: Yes indeed, these tests remain to be updated. These combine operations over multiple datasets from http://www.gis.usu.edu/~chrisg/python/2009/. Hence will take much more pruning to get the tests working again.

**Comment by pritamd47** [comment]
Thank you @visr for reviewing the work. Your comments are invaluable to me, while I try to get my head around how things work and the conventions of collaborating on an open-source project.

I have made some quick fixes based on your comments and have referenced the respective commits as comments.

Please let me know if I can improve in any way! Thank you for your time.

File: test/ospy/README.md [hunk 8]
```diff
@@ -1,3 +0,0 @@
-# README
-
-taken from http://www.gis.usu.edu/~chrisg/python/2009/
```
[the remark this answers is not in the document]
Comment (pritamd47) [reply test/ospy/README.md]: It is not removed, it still exists in the fork of the [ArchGDALDatasets](https://github.com/yeesian/ArchGDALDatasets) at [ArchGDALDatasets-Fork](https://github.com/pritamd47/ArchGDALDatasets). Adding this README.md file to remotefiles.jl might help

[the remark this answers is not in the document]
Comment (visr) [reply test/ospy/README.md]: Ah ok I didn't see that. Then it's fine I think.

File: test/test_ospy_examples.jl [hunk 9]
```diff
@@ -310,47 +310,47 @@ AG.read("ospy/data4/cropped_aster.img") do ds
...
+            count = 0
+            total = 0
+            buffer = Array{AG.pixeltype(band)}(undef, AG.blocksize(band)...)
+            for (cols,rows) in AG.windows(band)
+                AG.rasterio!(band, buffer, rows, cols)
+                data = buffer[1:length(cols),1:length(rows)]
+                count += sum(data .> 0)
+                total += sum(data)
+            end
+            @test total / count ≈ 86.88876013904982
```
Comment (pritamd47) [inline test/test_ospy_examples.jl]: This test should be similar to line 323 (which is the case in the original tests), but now "version 2" and "version 3" still fail for some reason. Might be due to some inner subtleties of ArchGDAL.jl which I am yet to learn about.

@visr Please review the changes and kindly look into the code to see if you can figure out what is going wrong. Thank you for your time and consideration.

**Comment by visr** [comment]
Hi @pritamd47. Sorry I didn't get to it earlier. I definitely want to have a reduced test size like you suggested in #104. As is, this PR currently changes 500 lines is a little unwieldy. This also makes it harder to review. To be able to quickly make some progress on the file sizes, I think it would be better if you create a new PR, which only removes the duplicate files, as a first step. I mentioned this on purpose in my comment at https://github.com/yeesian/ArchGDAL.jl/pull/106#pullrequestreview-359356199 because working in small chunks rather than large overhauls can actually end up being a faster way to progress, making incremental improvements. An added bonus is that you are less likely to get stuck, and will hopefully be more motivated to contribute more in the future. What do you think?

It would be better if you create a PR from a branch, rather than your master branch. That way I can also add commits if I need to fix up something, see also [these tips](https://github.com/JuliaLang/julia/blob/master/CONTRIBUTING.md#git-recommendations-for-pull-requests).

**Comment by pritamd47** [comment]
Thank you @visr for the suggestions. This indeed sounds like the right way forward, as I was getting overwhelmed with the number of changes needed to solve #104. I will start a new PR removing the duplicate files for starters.
The major takeaway for me would be, as someone who is just starting out with open source contribution, I should first start by tackling minor problems, and making small changes; otherwise, it is quite easy to get overwhelmed.

[This](https://github.com/JuliaLang/julia/blob/master/CONTRIBUTING.md#git-recommendations-for-pull-requests) link is a very nice resource, and I will now try to adhere to the recommendations listed here.

**Comment by yeesian** [comment]
Should this be closed in lieu of https://github.com/yeesian/ArchGDAL.jl/pull/121?

**Comment by pritamd47** [comment]
Yes 👍

**Pull request closed without merging**